index.js

```diff
@@ -91,6 +91,69 @@ say.speak = function(text, voice, speed, callback) {
   });
 };
 
+say.export = function(text, output, params, callback) {
+  var commands, pipedData;
+
+  if (!text) {
+    // throw TypeError because API was used incorrectly
+    throw new TypeError('Must provide text parameter');
+  }
+
+  if (!output || typeof output !== 'string') {
+    // throw TypeError because API was used incorrectly
+    throw new TypeError('Must provide output parameter');
+  }
+
+  if (typeof params !== 'object'){
+    params = {voice: 'Alex', speed: 175};
+  } else {
+    params.voice = params.voice || 'Alex';
+    params.speed = params.speed || 175;
+  }
+
+  if (typeof callback !== 'function') {
+    callback = function() {};
+  }
+
+  // tailor command arguments to specific platforms
+  if (process.platform === 'darwin') {
+    commands = ['-v', params.voice, text];
+
+    commands.push('-r', params.speed, '-o', output, '--data-format=LEF32@32000');
+
+  }  else {
+    // if we don't support the platform, callback with an error (next tick) - don't continue
+    return process.nextTick(function() {
+      callback(new Error('say.js export does not support platform ' + process.platform));
+    });
+  }
+
+  childD = spawn(say.speaker, commands);
+
+  childD.stdin.setEncoding('ascii');
+  childD.stderr.setEncoding('ascii');
+
+  if (pipedData) {
+    childD.stdin.end(pipedData);
+  }
+
+  childD.stderr.once('data', function(data) {
+    // we can't stop execution from this function
+    callback(new Error(data));
+  });
+
+  childD.addListener('exit', function (code, signal) {
+    if (code === null || signal !== null) {
+      return callback(new Error('say.js: could not talk, had an error [code: ' + code + '] [signal: ' + signal + ']'));
+    }
+
+    childD = null;
+
+    callback(null);
```

Comment:
Please provide the path to the file as the second argument to the callback when it successfully completes.

We should probably generate a random name to prevent overwriting existing files.